### PR TITLE
Make logging (to file) exception-free

### PR DESCRIPTION
**Pull request by dimitry-**
Switch from ofstream to FILE* to avoid accidental
exceptions while writing to log file.

Test: run orbit client - check logfile

### CODE REVIEW

File: src/OrbitBase/Logging.cpp [hunk 1]
```diff
@@ -35,22 +52,34 @@ ErrorMessageOr<void> TryRemoveOldLogFiles(const std::filesystem::path& log_dir)
...
-  if (log_file.is_open()) {
+  if (log_file != nullptr) {
     PLATFORM_ABORT();
   }
-  log_file.open(path, std::ofstream::out);
+
+  // TEMP_FAILURE_RETRY for FILE*
+  do {
+    log_file.reset(
+        fopen(path.c_str(), "wbe"));  // O_WRONLY, O_CLOEXEC for glibc, O_BINARY for windows
```
Comment (IrinaShkviro) [inline src/OrbitBase/Logging.cpp:67]: On Windows  'e' in "wbe" gives me "Invalid file open mode".

File: src/OrbitBase/Logging.cpp [hunk 2]
```diff
@@ -10,15 +10,32 @@
...
+#include "OrbitBase/UniqueResource.h"
+
+namespace orbit_base {
+
+#if !defined(TEMP_FAILURE_RETRY)
+// Windows does not have TEMP_FAILURE_RETRY, define it here
+#define TEMP_FAILURE_RETRY(expression)                 \
+  do {                                                 \
+    int64_t result = static_cast<int64_t>(expression); \
+  } while (result == -1 && errno == EINTR)
```
Comment (IrinaShkviro) [inline src/OrbitBase/Logging.cpp:30]:
```suggestion
#define TEMP_FAILURE_RETRY(expression)           \
  do {                                           \
    int64_t result;                              \
    do {                                         \
      result = static_cast<int64_t>(expression); \
    } while (result == -1 && errno == EINTR);    \
  } while (0)
#endif
```

**Review by IrinaShkviro — COMMENTED** [review]
Thank Dimitry! I left a couple of comments

File: src/OrbitBase/Logging.cpp [hunk 3]
```diff
@@ -10,15 +10,23 @@
...
+#include "OrbitBase/UniqueResource.h"
+
+namespace orbit_base {
 
 static absl::Mutex log_file_mutex(absl::kConstInit);
-std::ofstream log_file;
+static orbit_base::unique_resource log_file{static_cast<FILE*>(nullptr), [](FILE* f) {
+                                              if (f != nullptr) TEMP_FAILURE_RETRY(fclose(f));
+                                            }};
+// static std::ofstream log_file;
```
[the remark this answers is not in the document]
Comment (dimitry-) [reply src/OrbitBase/Logging.cpp:34]: done.

File: src/OrbitBase/Logging.cpp [hunk 4]
```diff
@@ -35,22 +43,34 @@ ErrorMessageOr<void> TryRemoveOldLogFiles(const std::filesystem::path& log_dir)
...
 
-void LogToFile(const std::string& message) {
+void LogToFile(const std::string& message) noexcept {
   absl::MutexLock lock(&log_file_mutex);
-  if (log_file.is_open()) {
-    log_file << message;
-    log_file.flush();
+  if (log_file != nullptr) {
+    // Ignore any errors that can happen, we cannot od anything about them at this
+    // point anyways.
```
[the remark this answers is not in the document]
Comment (dimitry-) [reply src/OrbitBase/Logging.cpp:77]: done.

File: src/OrbitBase/Logging.cpp (hunk 1 above)
Comment (dimitry-) [reply src/OrbitBase/Logging.cpp]: yep , add ifdef guard here too.

File: src/OrbitBase/Logging.cpp (hunk 2 above)
Comment (dimitry-) [reply src/OrbitBase/Logging.cpp]: removed this, now we have shortcut to expression instead.

**Review by IrinaShkviro — APPROVED** [review]
LGTM!

**Pull request merged**